src/github.com/mozilla-services/pushgo/simplepush/etcd_locator.go

```diff
@@ -179,11 +194,26 @@ func (l *EtcdLocator) Init(app *Application, config interface{}) (err error) {
 	}
 
 	l.closeWait.Add(2)
+	if l.startDelay > 0 {
+		l.closeWait.Add(1)
```

Comment:
This ensures `ReadyNotify()` always receives a value before the router is closed (i.e., the server is closed before it finishes starting up).